peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+feature sets—cascadelake and cannonlake are incomparable despite both
+descending from skylake, as each has unique AVX-512 extensions.
+
+**Implementation:** A language-agnostic JSON database stores
+microarchitecture metadata (features, compatibility relationships,
+compiler-specific optimization flags). Language bindings provide
+detection (queries ``/proc/cpuinfo``, matches to microarchitecture with
+largest compatible feature subset) and compatibility comparison
+operators.
+
+**Package Manager Integration:** Spack records target microarchitecture
+as package provenance (``spack install fftw target=broadwell``),
+automatically selects compiler flags, and enables
+microarchitecture-aware binary caching. The `European Environment for
+Scientific Software Installations (EESSI)
+<https://onlinelibrary.wiley.com/doi/full/10.1002/spe.3075>`__
+distributes optimized builds in separate subdirectories per
+microarchitecture (e.g., ``x86_64``, ``armv8.1a``, ``haswell``);
+runtime initialization uses ``archspec`` to select best compatible build
+when no exact match exists.
+
+
+Gentoo Linux
+''''''''''''
+
+`Gentoo Linux <https://www.gentoo.org>`_ is a source-first distribution
+with support for extensive package customization. This is primarily
+achieved via `USE flags
+<https://devmanual.gentoo.org/general-concepts/use-flags/index.html>`__:
+boolean flags exposed by individual packages and permitting fine-tuning
+the enabled features, optional dependencies and some build parameters
+(e.g. ``jpegxl`` for JPEG XL image format support,
+``cpu_flags_x86_avx2`` for AVX2 instruction set use). Flags can be
+toggled individually, and separate binary packages can be built for
+different sets of flags. The package manager can either pick a binary
+package with matching configuration or build from source.
+
+API and ABI matching is primarily done through use of `slotting
+<https://devmanual.gentoo.org/general-concepts/slotting/index.html>`__.
+Slots are generally used to provide multiple versions or variants of
+given package that can be installed alongside (e.g. different major GTK+
+or LLVM versions, or GTK+3 and GTK4 builds of WebKitGTK), whereas
+subslots are used to group versions within a slot, usually corresponding
+to the library ABI version. Packages can then declare dependencies bound
+to the slot and subslot used at build time. Again, separate binary
+packages can be built against different dependency slots. When
+installing a dependency version falling into a different slot or
+subslot, the package manager may either replace the package needing that
+dependency with a binary packages built against the new slot, or rebuild
+it from source.
+
+Normally, the use of slots assumes that upgrading to the newest version
+possible is desirable. When more fine-grained control is desired, slots
+are used in conjunction with USE flags. For example,
+``llvm_slot_{major}`` flags are used to select a LLVM major version to
+build against.
+
+
+Rationale
+=========
```

Comment:
I would move much of the prose from Specification to the Rationale. It took me a few read throughs of the Rationale section before concepts started to make sense. A couple of paragraphs or bullet points on how the design of the whole system fits together would be of great benefit at the start here

Reply:
Isn't the goal though, that people could actually implement the specification from reading "specification" section alone? Given the complexity of the proposal, I think it makes sense to include a more explicit introduction to all the concepts there.